test/functional/test_framework/test_framework.py

```diff
@@ -212,18 +223,24 @@ def parse_args(self):
             # It still needs to exist and be None in order for tests to work however.
             # So set it to None to force -disablewallet, because the wallet is not needed.
             self.options.descriptors = None
+            self.use_descriptors = OptionalBool()
```

Comment:
> Might be good to explain why this commit is needed. `None` will already force `-disablewallet`, and thus assert when a logic error occurs.

`self.use_descriptors` was needed because we can't be overriding `self.options.descriptors` in `main()` while also respecting its value from `parse_args()`. I also wanted to change the `self.options.descriptors` to an array of the wallet types to run with, which was incompatible with our current usage of that value.

`OptionalBool` was introduced to catch other logic errors within the tests themselves in areas that are not reliant on bitcoind running. For example, it caught the issue with the default wallet name.

> Also, it doesn't seem to be working, given that the tests fail when only sqlite is installed?

That's unrelated to this particular change. That was occurring because some tests add the wallet options since they can run with a wallet, but do not require it. For those tests, they were being run with both wallet types always, even if one type (e.g. legacy) was not available, causing the failure. I've pushed a fix for that.